### ISSUE TITLE
serde deserialization of structs containing unsigned integer types is failing

### ISSUE THREAD

**Issue by losfair** [losfair/wasm-core#2]
If a struct contains unsigned integer types like `u32` and `usize`, we get an error while trying to deserialize into it.

Serialization works fine and other types like `i32`, `String`, `Vec` and nested structs work fine.

Code:

```rust
#[derive(Serialize, Deserialize)]
struct Foo {
    ivalue: usize
}

#[no_mangle]
pub extern "C" fn ser_de(v: i32) -> i32 {
    let foo = Foo {
        ivalue: v as usize
    };
    let ser = serde_json::to_string(&foo).unwrap();
    let de: Foo = serde_json::from_str(&ser).unwrap();
    de.ivalue as i32
}
```

Error:

```
Error: UnreachableExecuted
Backtrace:
Backtrace { frames: [BtFrame { name: Some("__rust_start_panic") }, BtFrame { name: Some("rust_panic") }, BtFrame { name: Some("_ZN3std9panicking20rust_panic_with_hook17h4ef656543b7370b7E") }, BtFrame { name: Some("_ZN3std9panicking15begin_panic_fmt17h14153e6c183bf10cE") }, BtFrame { name: Some("rust_begin_unwind") }, BtFrame { name: Some("_ZN4core9panicking9panic_fmt17hacb4853db78127fcE") }, BtFrame { name: Some("_ZN4core6result13unwrap_failed17h80cf1f550623fd5aE") }, BtFrame { name: Some("ser_de") }] }
```

**Comment by losfair** [comment]
Signed `Load` instructions seem to be implemented incorrectly, which might be the cause of this issue.

**Comment by losfair** [comment]
The same code works fine while running on top of the JIT backend.

**Comment by losfair** [comment]
Fixed in `d609f498`.
